### PR TITLE
#19270 added elasticsearch provisioning

### PR DESCRIPTION
**Pull request by daniel-jeremia**
## Changes
- Added ECK Operator installation as subchart
- Added elasticsearch and kibana provisioning
- Added post install hooks to some service to make sure it runs after initialization

### CODE REVIEW

File: charts/datasaur/templates/jobs/migrator-job.yaml [hunk 1]
```diff
@@ -10,7 +10,19 @@ spec:
...
       labels:
         app: {{ .Release.Name }}-migrator
+      annotations:
+        "helm.sh/hook": post-install
+        "helm.sh/hook-weight": "1"
     spec:
+      initContainers:
+        - name: wait-for-mariadb-before-starup
+          image: busybox
+          command:  ["sh", "-c", "until nc -z {{ .Release.Name }}-mariadb-service 3306 > /dev/null; do echo Waiting for mariadb.; sleep 2; done;"]
```
Comment (tonoman3g) [inline charts/datasaur/templates/jobs/migrator-job.yaml]: does this work for RDS or managed service?

File: charts/datasaur/templates/main/backend-worker-critical-deployment.yaml [hunk 2]
```diff
@@ -5,6 +5,9 @@ metadata:
   labels:
     app: {{ .Release.Name }}-backend-critical-worker
   name: {{ .Release.Name }}-backend-critical-worker
+  annotations:
+    "helm.sh/hook": post-install
+    "helm.sh/hook-weight": "5"
```
Comment (tonoman3g) [inline charts/datasaur/templates/main/backend-worker-critical-deployment.yaml:10]: what is the meaning of hook-weight 5? 🤔

File: charts/datasaur/templates/start/elasticsearch-eck.yaml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+{{- if .Values.elasticsearch.enabled }}
+apiVersion: elasticsearch.k8s.elastic.co/v1
+kind: Elasticsearch
+metadata:
+  name: {{ .Release.Name }}-elasticsearch
+  annotations:
+    "helm.sh/hook": post-install,post-upgrade
+spec:
+  version: 8.10.4
```
Comment (tonoman3g) [inline charts/datasaur/templates/start/elasticsearch-eck.yaml:9]: is this what you did to override the helm chart and ES version?

File: charts/datasaur/templates/start/elasticsearch-eck.yaml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  http:
+    tls:
+      selfSignedCertificate:
+        disabled: true
+  nodeSets:
+  - name: master-nodes
+    count: 3
```
Comment (tonoman3g) [inline charts/datasaur/templates/start/elasticsearch-eck.yaml:16]: is the minimum node for ES running well 3?

File: charts/datasaur/templates/start/elasticsearch-eck.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+      node.roles: ["master", "data" ,"data_content" ,"data_hot", "data_warm", "data_cold", "data_frozen", "ml", "ingest", "remote_cluster_client", "transform"]
+    volumeClaimTemplates:
+    - metadata:
+        name: elasticsearch-data
+      spec:
+        accessModes:
+        - ReadWriteOnce
+        resources:
+          requests:
+            storage: 100Gi
```
Comment (tonoman3g) [inline charts/datasaur/templates/start/elasticsearch-eck.yaml]: can we override the storage?
or should we expose so the storage can be overriden?

File: charts/datasaur/templates/jobs/migrator-job.yaml (hunk 1 above)
Comment (daniel-jeremia) [reply charts/datasaur/templates/jobs/migrator-job.yaml]: ah iya juga ya, harus diganti brati jadi DATABASE_HOST

File: charts/datasaur/templates/main/backend-worker-critical-deployment.yaml (hunk 2 above)
Comment (daniel-jeremia) [reply charts/datasaur/templates/main/backend-worker-critical-deployment.yaml:10]: itu priority, the less weight the more priority
aku set 5 karna mau backend/worker jalan belakangan @tonoman3g

File: charts/datasaur/templates/start/elasticsearch-eck.yaml (hunk 3 above)
Comment (daniel-jeremia) [reply charts/datasaur/templates/start/elasticsearch-eck.yaml:9]: no.. ini buat yang self-hosted aku udah pake ECK, jdi ini udah latest and recommended approach from elastic @tonoman3g

File: charts/datasaur/templates/start/elasticsearch-eck.yaml (hunk 4 above)
Comment (daniel-jeremia) [reply charts/datasaur/templates/start/elasticsearch-eck.yaml:16]: yes, karna master nya harus ganjil dan lebih besar dari 1, jadi ya minimal 3 @tonoman3g

File: charts/datasaur/templates/start/elasticsearch-eck.yaml (hunk 5 above)
Comment (daniel-jeremia) [reply charts/datasaur/templates/start/elasticsearch-eck.yaml]: yes we can, boleh juga sih di expose @tonoman3g

Comment (tonoman3g) [reply charts/datasaur/templates/start/elasticsearch-eck.yaml]: boleh deh diexpose

File: charts/datasaur/templates/jobs/migrator-job.yaml (hunk 1 above)
Comment (daniel-jeremia) [reply charts/datasaur/templates/jobs/migrator-job.yaml]: decided to remove this and fallback to retries, best practice nya ternyata begitu.

**Review by tonoman3g — APPROVED** [review]

**Pull request merged**